src/data_device/offer.rs

```diff
@@ -106,6 +106,29 @@ impl DataOffer {
         Ok(unsafe { FromRawFd::from_raw_fd(readfd) })
     }
 
+    /// Receive data to the write end of a raw file descriptor. If you have the read end, you can read from it.
+    ///
+    /// You can do this several times, as a reaction to motion of
+    /// the dnd cursor, or to inspect the data in order to choose your
+    /// response.
+    ///
+    /// Note that you should *not* read the contents right away in a
+    /// blocking way, as you may deadlock your application doing so.
+    /// At least make sure you flush your events to the server before
+    /// doing so.
+    ///
+    pub unsafe fn receive_to_fd(&self, mime_type: String, writefd: i32) -> std::io::Result<()> {
+        use nix::unistd::close;
+
+        self.offer.receive(mime_type, writefd);
+
+        if let Err(err) = close(writefd) {
+            log::warn!("Failed to close write pipe: {}", err);
+        }
+
+        Ok(())
+    }
```

Comment:
Is `std::io::Result<()>` the right return type for this? The method always returns `Ok(())`

Reply:
Oh good point, my bad. I'll fix that.